### PR TITLE
fix(table): slot, bodyData, events improvements

### PR DESCRIPTION
**Pull request by theJohnnyMe**
Fixing:

- slot names
- add/remove event listeners logic
- updating new and removed props
- changing logic for adding bodyData

### CODE REVIEW

File: src/components/Table/SearchTable.tsx [hunk 1]
```diff
@@ -1,46 +1,60 @@
+import { useEffect, useRef } from 'react';
+import exampleData from './example-data.json';
+
 interface SearchTableProps {
   cancelEvent?: boolean;
 }
 
 const SearchTable = ({ cancelEvent }: SearchTableProps) => {
+  const searchTable = useRef<HTMLTdsTableElement>(null);
```
Comment (nathalielindqvist) [inline src/components/Table/SearchTable.tsx:9]: Looks like both the searchable Tables are filtered when searching in the first Table:
image

File: src/components/Table/BatchActionTable.tsx [hunk 2]
```diff
@@ -1,28 +1,35 @@
...
+        setData(table.current.getAttribute('data-selected-rows') as string);
       }
       modal.current?.showModal();
     }
   };
   return (
     <>
       <tds-modal ref={modal} id="my-modal" size="lg" actions="static">
-        <h5 className="tds-modal-headline" slot="tds-modal-headline">
+        <h5 className="tds-modal-headline" slot="header">
```
Comment (nathalielindqvist) [inline src/components/Table/BatchActionTable.tsx:26]: When selecting the top most Checkbox in the Batch Action Table all the rows are not selected (as I believe is the expected behavior, correct me if that's not the case).
image

File: src/components/Table/SearchTable.tsx (hunk 1 above)
Comment (alexandereneroth) [reply src/components/Table/SearchTable.tsx:9]: thought: they might shade an ID or something like that

File: src/components/Table/BatchActionTable.tsx (hunk 2 above)
Comment (theJohnnyMe) [reply src/components/Table/BatchActionTable.tsx]: I have rebased the branch and now the issues are gone.

File: src/components/Table/SearchTable.tsx (hunk 1 above)
Comment (theJohnnyMe) [reply src/components/Table/SearchTable.tsx:9]: I have rebased the branch and now the issues are gone. Will keep my eyes while doing future work.

**Pull request merged**